Samples_for_Extending_Pentaho/Reference Implementations/Security/SAML 2.0/pentaho-saml/src/main/java/org/pentaho/platform/spring/security/saml/logout/PentahoSamlLogoutProcessingFilter.java

```diff
@@ -26,26 +32,30 @@ public PentahoSamlLogoutProcessingFilter( LogoutSuccessHandler logoutSuccessHand
   }
 
   @Override
-  public void processLogout(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws
-      IOException, ServletException {
+  public void processLogout( HttpServletRequest request, HttpServletResponse response, FilterChain chain )
+    throws IOException, ServletException {
 
-    if( requiresLogout( request, response ) ){
+    if ( requiresLogout( request, response ) ) {
 
       try {
 
-        Authentication auth = Utils.getAuthenticationFromRequest( request );
+        Authentication auth = Utils.getAuthenticationFromRequest( request, this.requireProxyWrapping );
 
-        if( auth != null ) {
+        if ( auth != null ) {
           SecurityContextHolder.getContext().setAuthentication( auth );
         }
 
       } catch ( NoSuchMethodException | InvocationTargetException | IllegalAccessException | ProxyException e ) {
-        logger.error( e.getMessage(), e );
+        LOGGER.error( e.getMessage(), e );
       }
     }
 
     super.processLogout( request, response, chain );
 
   }
 
+  public void setRequireProxyWrapping( boolean requireProxyWrapping ) {
+    this.requireProxyWrapping = requireProxyWrapping;
+  }
+
```

Comment:
Not a blocker: 

Add a getter as well; then use it at line

```Utils.getAuthenticationFromRequest( request, this.requireProxyWrapping )``` instead of a direct call to the class attribute. Doing so will ease unit testing, as well as allowing for ( potential future ) subclasses of `PentahoSamlLogoutProcessingFilter` to override it, should they have the need to.

Other than that, I'm 👍 with the PR